src/sagemaker/serve/builder/model_builder.py

```diff
@@ -779,7 +763,37 @@ def build(  # pylint: disable=R0911
                 else:
                     return self._build_for_transformers()
 
-        self._build_validations()
+        # Set TorchServe as default model server
+        if not self.model_server:
+            self.model_server = ModelServer.TORCHSERVE
+            return self._build_for_torchserve()
+
+        raise ValueError("%s model server is not supported" % self.model_server)
+
+    def _build_validations(self):
+        """Validations needed for model server overrides, or auto-detectection or fallback"""
+        if self.mode == Mode.IN_PROCESS:
+            raise ValueError("IN_PROCESS mode is not supported yet!")
+
+        if self.inference_spec and self.model:
+            raise ValueError("Cannot have both the Model and Inference spec in the builder")
+
+        if self.image_uri and not is_1p_image_uri(self.image_uri) and self.model_server is None:
+            raise ValueError(
+                "Model_server must be set when non-first-party image_uri is set. "
+                + "Supported model servers: %s" % supported_model_server
+            )
+
+    def _build_for_model_server(self):  # pylint: disable=R0911, R1710
+        """Model server overrides"""
+        if self.model_server not in supported_model_server:
+            raise ValueError(
+                "%s is not supported yet! Supported model servers: %s"
+                % (self.model_server, supported_model_server)
+            )
+
+        if not self.model and not self.model_metadata:
+            raise ValueError("Missing required parameter `model` or 'ml_flow' path")
```

Comment:
should this check for `model_path` too? also nit: either change to `mlflow_path` or remove reference since in the future there are other options too such as `fine_tuning_model_path`

Reply:
addressed